pkg/kubectl/cmd/config/view.go

```diff
@@ -93,9 +93,9 @@ func NewCmdConfigView(f cmdutil.Factory, out, errOut io.Writer, ConfigAccess cli
 	options.Merge.Default(true)
 	mergeFlag := cmd.Flags().VarPF(&options.Merge, "merge", "", "Merge the full hierarchy of kubeconfig files")
 	mergeFlag.NoOptDefVal = "true"
-	cmd.Flags().BoolVar(&options.RawByteData, "raw", false, "Display raw byte data")
-	cmd.Flags().BoolVar(&options.Flatten, "flatten", false, "Flatten the resulting kubeconfig file into self-contained output (useful for creating portable kubeconfig files)")
-	cmd.Flags().BoolVar(&options.Minify, "minify", false, "Remove all information not used by current-context from the output")
+	cmd.Flags().BoolVar(&options.RawByteData, "raw", options.RawByteData, "Display raw byte data")
```

Comment:
I tried `kubectl config view` and `kubectl config view --raw`, it runs well. What's the problem? :dizzy_face:

Reply:
> I tried kubectl config view and kubectl config view --raw, it runs well. What's the problem?

I have a detailed explanation here: https://github.com/kubernetes/kubernetes/pull/60621#issuecomment-369589621, but binding to structs in an composeable way makes it possible to write more reliable unit tests with predictable defaulting behavior and for other flags (not this flag in particular) allows for later composition.

I think the burden runs the other way around.  We should not choose build our code with multiple stages of defaulting without a clear benefit.  Momentum does not meet that bar, especially with contributors willing to help scrub the code so that we can later reap the benefits.

@charrywanganthony I'm not familiar with the face. What are the dead eyes for?  This pull addresses a real need in our codebase and was made and reviewed in good faith.

Reply:
It's a dizzy face. thx for your time.